src/sna/sna_display.c

```diff
@@ -5052,6 +5052,8 @@ sna_set_cursor_position(ScrnInfoPtr scrn, int x, int y)
 				arg.handle = cursor->handle;
 			}
 
+			arg.x += xu;
+			arg.y += yu;
```

Comment:
Is there any reason these need to be modified in the else case? I assume not since there's a `goto disable` that bypasses the adjustment above, but I don't fully understand when arg is used.